src/AppScopeTrait.php

```diff
@@ -63,18 +63,6 @@ trait AppScopeTrait
      */
     public $unique_hashes = [];
 
-    private function assertInstanceOfApp(object $app): void
```

Comment:
this is there for a reason, you want to assing app that is not `\Atk4\Ui\App` based?

Reply:
yes of course, why not? I use the AppScope without using Atk4/Ui, I use it to centralize service in a master class as an App and use even Container + Trackable + Initializer + Hook without any problem to apply a composition pattern and in any element have a reference to a master class which is well organized and not static

Reply:
this is problematic because of hinted type for static analysis

I think we need to move this trait to `atk4/ui` or make another trait based on this one there

but what is the reason for it then in core?

Reply:
I know traits gives that type of problem, because Traits are wrong, private methods in Traits are much wrong.
Like i said in private, i like the work you are doing and the direction of "be much strict" but not with traits, be much strict using traits is wrong.
If you want to go further in this direction why not start thinking about using interfaces?